packages/cadl-ranch-specs/http/azure/core/basic/main.tsp

```diff
@@ -214,3 +216,41 @@ Expected response body:
 ```
 """)
 op export is ResourceOperations.ResourceAction<User, UserExportParams, User>;
+
+model UserList {
+  collection: User[];
+}
+
+@scenario
+@doc("Exports all users")
+@summary("Exports all users.")
+@scenarioDoc("""
+Should generate a model named User.
+
+Expected query parameter: format=json
+Expected query parameter: api-version=2022-12-01-preview
+
+Expected response body:
+```json
+[
+  {
+    "id": 1,
+    "name": "Madge",
+    "etag": "11bdc430-65e8-45ad-81d9-8ffa60d55b59"
+  },
+  {
+    "id": 2,
+    "name": "John",
+    "etag": "22bdc430-65e8-45ad-81d9-8ffa60d55b59"
+  }
+]
+""")
+@collectionAction(User, "exportallusers")
+op exportAllUsers is ResourceOperations.ResourceCollectionAction<
```

Comment:
Why do you need `@collectionAction(User, "exportallusers")`? I think it should be included in `ResourceCollectionAction` operation template.

See if you can add a `@post` above the operation. This would be redundant, but the summary seems messed-up without it https://github.com/Azure/cadl-ranch/pull/625/files#diff-e96e1cba09706c8a0e00bde73f721986543b99e3a0db21b3bbb98559db31185a

This is not required. But the summary seems to be

Reply:
Based on the generated results, it is necessary to retain @collectionAction. After removing it, the generated path directly calls the operation name, resulting in uppercase letters appearing in the path. Therefore, I added @collectionAction to prevent this from happening. @post added.

Reply:
Eh, who says uppercase letters cannot appear in the path?

Reply:
In a previous PR, there was a comment suggesting that I change the uppercase letters in the path to lowercase. Additionally, I have rarely seen paths with uppercase letters in mockapi. So, which approach is more appropriate?

Reply:
https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#http-url-casing

I don't have particular preference in this case. It is fine for all-lower-case. Just want to know why the extra `@collectionAction`.